### PR TITLE
podman_quadlet: Check for podman version

### PR DESCRIPTION
**Pull request by rbmarliere**
This test makes use of a feature only available for podman > 5.2.0.

- Related ticket: https://progress.opensuse.org/issues/173764
- Verification run:
http://rmarliere-openqa.qe.prg2.suse.org/tests/208 (Leap Micro 6.1 with podman 5.0.3)
https://openqa.opensuse.org/tests/4686249 (TW with podman 5.3.1)

### CODE REVIEW

File: tests/containers/podman_quadlet.pm [hunk 1]
```diff
@@ -136,12 +137,25 @@ sub run {
     systemctl("disable --now $unit_name.service");
     systemctl("stop $unit_name-volume.service");
     systemctl("stop $unit_name-network.service");
-    systemctl("stop $unit_name-build.service");
+    systemctl("stop $unit_name-build.service", ignore_failure => 1);
```
Comment (grisu48) [inline tests/containers/podman_quadlet.pm:140]: Why is this needed?

File: tests/containers/podman_quadlet.pm [hunk 2]
```diff
@@ -98,30 +97,32 @@ sub run {
...
-    systemctl("start $unit_name-build.service", timeout => 180);
-    record_info('Build output', script_output("journalctl --no-pager -u $unit_name-build"));
-
-    systemctl("is-active $unit_name-build.service", expect_false => 1);
-    validate_script_output('podman images -n', qr/$build_imagetag/);
+    if (check_min_runtime_version('5.2.0')) {
+        script_retry("podman pull $src_image", retry => 3, delay => 60, timeout => 180);
+        systemctl("start $unit_name-build.service", timeout => 180);
+        record_info('Build output', script_output("journalctl --no-pager -u $unit_name-build"));
+        systemctl("is-active $unit_name-build.service", expect_false => 1);
```
Comment (grisu48) [inline tests/containers/podman_quadlet.pm:116]: We need to check if this unit has run, not if it is active. This check will always pass.

File: tests/containers/podman_quadlet.pm (hunk 1 above)
Comment (rbmarliere) [reply tests/containers/podman_quadlet.pm:140]: It's either that or

```perl
if (check_min_runtime_version('5.2.0')) {
    systemctl("stop $unit_name-build.service");
}
```

File: tests/containers/podman_quadlet.pm (hunk 2 above)
Comment (rbmarliere) [reply tests/containers/podman_quadlet.pm:116]: But after starting it wouldn't be active if there was an error. Also, this is what the test had before, I only added the if statement.

File: tests/containers/podman_quadlet.pm (hunk 1 above)
Comment (grisu48) [reply tests/containers/podman_quadlet.pm:140]: Suggestion: Introduce `my $has_build = check_min_runtime_version('5.2.0')` at the top and then something like

```
systemctl("stop $unit_name-build.service") if($has_build);
```

Might also help to avoid running this check multiple times.

File: tests/containers/podman_quadlet.pm (hunk 2 above)
Comment (grisu48) [reply tests/containers/podman_quadlet.pm:116]: Ah ok, then let's leave it for now.

File: tests/containers/podman_quadlet.pm (hunk 1 above)
Comment (rbmarliere) [reply tests/containers/podman_quadlet.pm]: Ah good idea, fixed in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/20751/commits/f4b18942cae36b2a523995274024bbc3e33bb79d

**Review by ricardobranco777 — APPROVED** [review]
